pkgs/development/tools/misc/dura/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, fetchFromGitHub, rustPlatform, pkgs }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dura";
+  version = "unstable-2022-01-07";
+
+  src = fetchFromGitHub {
+    owner = "tkellogg";
+    repo = "dura";
+    rev = "a41863879772650f2538a978ca926357bae674db";
```

Comment:
👍 this rev fixed a big bug. There's no more huge bugs that i'm aware of, but i had wanted to let it bake for a little while before giving anybody any notion that it's stable. With that said, I don't mind if this package is merged, as long as there's no expectation that it's stable.

Reply:
Excellent, thanks !

By the way, I had to disable the tests, for some reasons it was not working. Do you think you could have a look at it ?

Reply:
Here's the complete log: https://gist.github.com/drupol/394fc75a6e8ba5647e6b895608f01eae

Reply:
I think the problem is that the tests [want `target/debug/dura`](https://github.com/tkellogg/dura/blob/f18bc40ccae91e5efa1ea6e183d9b8adef100f64/tests/util.rs#L98), but we won't have that.

We *can* force testing in debug mode by adding `checkType = "debug";`, which will cause the builder to build both in release and debug, but with Nixpkgs, our output is in `target/<triple>/debug/` instead (eg `target/x86_64-unknown-linux-gnu/debug/`), so it won't work either.

Reply:
Will `$CARGO_BIN_EXE_DURA` be set? [docs](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates)

Reply:
I fixed this in https://github.com/tkellogg/dura/commit/67650f9d696985e9b93394e5ec581e5f771184dd. See if that works

Reply:
I think it improved things, but there is still an error: https://gist.github.com/drupol/5c7559580688546156e3f79b6a43c386

Reply:
Ah, for that test I went outside the normal behavior and neglected to create the parent directory. Unsuprisingly, Mac works different than NixOS regarding tempdirs.

Try this commit: `845b5f545c162ab65871b5869fa1a90dca4fe3ac`

Reply:
Doing it now.

Reply:
The failure is still there: https://github.com/NixOS/nixpkgs/pull/153803/checks?check_run_id=4748654482

Reply:
The logfile: https://gist.github.com/drupol/70a2a0208b216df9ad6209ca4f06d009

Reply:
That seems to have been an upstream problem with tests (see tkellogg/dura#54), although fixing that still leaves unrelated tests failing.